File: bin/people-array.js
'use strict';

const loadPeople = require('../lib/people.js');

loadPeople().then((people) => {
  let countOlderThanHenry = people.reduce((count, person) =>
    person.age() > 54 ? count + 1 : count,
    0);
  let countFirstLast = people.reduce((count, person) =>
    person.surname[0] === person.given_name[0] ? count + 1 : count,
    0);
  let averageAge = people.reduce((ages, person) =>
    ages + person.age(),
    0) / people.length;
  console.log(`People older than Henry: ${countOlderThanHenry}`);
  console.log(`People First/Last: ${countFirstLast}`);
  console.log(`Average age: ${averageAge}`);
}).catch((error) => console.log(error));

File: bin/people_array.rb
require_relative '../lib/people.rb'

people = People.new.people
people_older_than_henry = people.count { |p| p.age > 54 }
people_first_last = people.count { |p| p.surname[0] == p.given_name[0] }
average_age = people.reduce(0) { |a, e| a + e.age } / people.length.to_f

puts "People older than Henry: #{people_older_than_henry}"
puts "People first/last: #{people_first_last}"
puts "Average age: #{average_age}"
